### ISSUE TITLE
[Bug] Undo button flashes on a different color after opening menu

### ISSUE THREAD

**Issue by BrayanDSO** [ankidroid/Anki-Android#10697]
###### Reproduction Steps

0. Have undo on the app bar (I haven't tested with other buttons)
1. Go review a card
2. Tap the menu button

https://user-images.githubusercontent.com/69634269/161448674-7793971f-d74f-476c-a789-b5b439500889.mp4

###### Expected Result

Undo button color is consistent

###### Actual Result

Undo button color changes

###### Debug info
```
AnkiDroid Version = 2.16alpha55

Android Version = 11

Manufacturer = samsung

Model = SM-N770F

Hardware = exynos9810

Webview User Agent = Mozilla/5.0 (Linux; Android 11; SM-N770F Build/RP1A.200720.012; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/99.0.4844.88 Mobile Safari/537.36

ACRA UUID = 037bbc86-38a4-4ad7-9c42-05870e9b6086

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true
```

###### Research
*Enter an [x] character to confirm the points below:*

- [X] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] I have searched for similar existing issues here and on the user forum
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### More context
![2022_04_03-18_05_35](https://user-images.githubusercontent.com/69634269/161448717-d1ece6ae-4c59-48ce-8229-cce978fe947c.png)
![2022_04_03-18_05_50](https://user-images.githubusercontent.com/69634269/161448718-4d3b2712-6d35-4313-8a8a-4bc0f77e78f4.png)

**Comment by BrayanDSO** [comment]
Happens on all 4 themes

**Comment by BrayanDSO** [comment]
In the same `app bar icons color` theme, I believe `Reschedule` icon should be the same color of the others

![2022_04_03-18_07_26](https://user-images.githubusercontent.com/69634269/161448768-498b7173-fe84-400d-972e-14a325dd787f.png)

**Comment by shubham419** [comment]
hi I would like to work on  the issue

**Comment by Umesh-01** [comment]
@BrayanDSO I want to work on this issue, can you tell me in which branch should I make these changes??

**Comment by Umesh-01** [comment]
> Fork AnkiDroid repository. Make your changes on any branch _**on your fork**_, then make a pull request to AnkiDroid main when it is finished

Okk, thank you !

**Comment by Umesh-01** [comment]
@BrayanDSO I think @shubham419 is not working on this, should I go ahead with this issue??

**Comment by mikehardy** [comment]
This is open source which means people have no obligation to submit to time pressure and sometimes real work happens, but not fast. Give @shubham419 24 hours to reply, then if by tomorrow there is no reply we can consider it open again

**Comment by Umesh-01** [comment]
Sure @mikehardy

**Comment by Umesh-01** [comment]
> This is open source which means people have no obligation to submit to time pressure and sometimes real work happens, but not fast. Give @shubham419 24 hours to reply, then if by tomorrow there is no reply we can consider it open again

@mikehardy may I work on this now??

**Comment by mikehardy** [comment]
Yes! Good luck :-)

**Comment by Umesh-01** [comment]
@mikehardy can you please guide me in which package or file should I look for this bug, I tried but didn't find this issue yet.

**Comment by david-allison** [comment]
> @mikehardy can you please guide me in which package or file should I look for this bug, I tried but didn't find this issue yet.

I'd start with undo():

https://github.com/ankidroid/Anki-Android/blob/1a453679e62cabed8afe17ae4817d2ceeeae0d5e/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt#L806-L821

**Comment by mikehardy** [comment]
Probably related - there appears to be some deeply-rooted problem with order of operations on the app bar / menu build:
#10792
#10785 
#10732 
#10691 
#10683 
#10445

**Comment by winstoncooke** [comment]
> Probably related - there appears to be some deeply-rooted problem with order of operations on the app bar / menu build: #10792 #10785 #10732 #10691 #10683 #10445
> 
> If someone wanted to _really_ figure out what exactly was happening with the order of operations here and fix all of them that would be fantastic as I think they are all related. Might be able to close 7 issues in one PR.
> 
> Some interaction between Activity onCreate / onCollectionLoaded / prefs load (for color theme) invalidate menu / rebuild menu thinking

I've been looking into this when I have time away from my gsoc application. The satisfaction from closing 7 PRs would be immeasurable! I'll continue updating if I make further progress.

**Comment by BrayanDSO** [comment]
Solved with #11061
